nemo/collections/nlp/models/language_modeling/megatron_lm_encoder_decoder_model.py

```diff
@@ -743,7 +896,128 @@ def predict_step(self, batch: Any, batch_idx: int, dataloader_idx: Optional[int]
         logging.info(f"response: {response}")
         return response
 
-    def decode(self, tokens_enc, enc_mask, num_tokens_to_generate, encoder_input=None, tokenizer=None):
+    def encode(self, tokens_enc, enc_mask, encoder_input=None, reconfigure_microbatch=True):
+        """
+        tokens_enc - encoder input tokens 
+        enc_mask - corresponding mask
+        encoder_input - encoder input (bypass tokens), if given tokens_enc can be None.
+        """
+        # Check whether the DDP is initialized. This is needed when running inference outside of training loop.
+        if parallel_state.is_unitialized():
```

Comment:
Can we move 906 - 945 to a common function that is called by encode() and decode()

Reply:
Not identical to decode. Let's discuss.